deep_quoridor/src/quoridor_env.py

```diff
@@ -434,50 +440,78 @@ def _next_player(self):
     def get_opponent(self, agent):
         return "player_1" if agent == "player_0" else "player_0"
 
+    def get_goal_row(self, agent):
+        return 0 if agent == "player_1" else self.board_size - 1
+
+    def _dfs(self, row, col, target_row, visited, any_path=True):
+        """
+        Performs a depth-first search to find whether the pawn can reach the target row.
+
+        Args:
+            row (int): The current row of the pawn
+            col (int): The current column of the pawn
+            target_row (int): The target row to reach
+            visited (numpy.array): A 2D boolean array with the same shape as the board,
+                indicating which positions have been visited
+            If any_path is set to true, the first path to the target row will be returned (faster).
+            Otherwise, the shortest path will be returned (potentially slower)
+
+        Returns:
+            int: Number of steps to reach the target or -1 if it's unreachable
+        """
+        if row == target_row:
+            return 0
+
+        visited[row, col] = True
+
+        # Find out the forward direction to try it first and maybe get to the target faster
+        fwd = 1 if target_row > row else -1
+
+        moves = [(row + fwd, col), (row, col - 1), (row, col + 1), (row - fwd, col)]
+        best = -1
+        for new_row, new_col in moves:
+            if (
+                self._is_in_board(new_row, new_col)
+                and not self.is_wall_between(row, col, new_row, new_col)
+                and not visited[new_row, new_col]
+            ):
+                dfs = self._dfs(new_row, new_col, target_row, visited)
+                if dfs != -1:
+                    if any_path:
+                        return dfs + 1
+                    if best == -1 or dfs + 1 < best:
+                        best = dfs + 1
+
+        return best
+
+    def can_reach(self, row, col, target_row, any_path=True):
```

Comment:
now that it returns an int rather than a bool, can_reach is not a good name.  I think similarly to Euclidean Distance or Hamiltonean Distance, this should be named Julianian Distance 🤣 
You could leave the `can_reach` method and make it return boolean, and add another method julianian_distance